### PR TITLE
Fixes #67

### LINKED ISSUE #67: rbx_xml empty BinaryString values should not include CDATA
Right now they serialize like this:

```xml
<BinaryString name="Tags"><![CDATA[]]></BinaryString>
```

but they should serialize like this instead:

```xml
<BinaryString name="Tags"></BinaryString>
```

This is safe because it lines up with how Roblox serializes these values.

### PR DESCRIPTION
**Pull request by Kampfkarren**
Fixes #67.

### CODE REVIEW

**Comment by LPGhatguy** [comment]
Will this still produce CDATA for non-empty binary strings?

**Comment by LPGhatguy** [comment]
The round trip tests just make sure that `x == deserialize(serialize(x))`. Since I bet we'll treat them the same when _deserializing_, there isn't really any verification that the serialization stays consistent...

I opened #77 to track that fix in the future.

**Comment by Kampfkarren** [comment]
Fair enough. I'll write a test for the binary thing later today.

**Comment by Kampfkarren** [comment]
Oh shoot, I completely misread the actual issue this was fixing 😅 
I'll fix it up in a minute.

File: rbx_xml/src/types/binary_string.rs [hunk 1]
```diff
@@ -20,7 +20,9 @@ impl XmlType<[u8]> for BinaryStringType {
         value: &[u8],
     ) -> Result<(), EncodeError> {
         writer.write(XmlWriteEvent::start_element(Self::XML_TAG_NAME).attr("name", name))?;
-        writer.write(XmlWriteEvent::cdata(&base64::encode(value)))?;
+        if value.len() > 0 {
```
Comment (LPGhatguy) [inline rbx_xml/src/types/binary_string.rs]:
```suggestion
        if !value.is_empty() {
```

I think there's a Clippy lint that talks about this, but alas, they're not as warnings on this repo yet, heh.

**Comment by LPGhatguy** [comment]
Can you add an entry to the changelog for `rbx_xml` and link to this PR?

**Comment by LPGhatguy** [comment]
Travis build hung, I'll restart it...

**Pull request merged**